nixos/modules/services/networking/lxd-image-server.nix

```
@@ -0,0 +1,142 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.lxd-image-server;
+
+  location = "/var/www/simplestreams";
+in
+{
+  options = {
+    services.lxd-image-server = {
+      enable = mkEnableOption "lxd-image-server";
+
+      group = mkOption {
+        type = types.str;
+        description = "Group assigned to the user and the webroot directory";
+        default = "nginx";
+        example = "www-data";
+      };
+
+      extraConfig = mkOption {
+        type = with types; lines;
+        description = ''
+          Configuration for lxd-image-server
+
+          Example see https://github.com/Avature/lxd-image-server/blob/master/config.toml
+        '';
+        default = "";
+      };
+
+      nginx = {
+        enable = mkEnableOption "nginx";
+        domain = mkOption {
+          type = types.str;
+          description = "Domain to use for nginx virtual host";
+          example = "images.example.org";
+        };
+      };
+    };
+  };
+
+  config = mkMerge [
+    (mkIf (cfg.enable) {
+      users.users.lxd-image-server = {
+        isSystemUser = true;
+        group = cfg.group;
+      };
+      users.groups.${cfg.group} = {};
+
+      environment.etc."lxd-image-server/config.toml".text = cfg.extraConfig;
+
+      services.logrotate.extraConfig = ''
+        /var/log/lxd-image-server/lxd-image-server.log {
+            daily
+            missingok
+            rotate 21
+            compress
+            delaycompress
+            copytruncate
+            notifempty
+        }
+      '';
```

```suggestion
      services.logrotate.paths.lxd-image-server = {
        path = "/var/log/lxd-image-server/lxd-image-server.log";
        frequency = "daily";
        keep = 21;
        extraConfig = ''
          missingok
          compress
          delaycompress
          copytruncate
          notifempty
        '';
  };
```

Why? By using `extraConfig` instead of `paths` the user is completely unable to tweak log rotation.